src/client/public/scripts/components/nav-element-input.ts

```diff
@@ -0,0 +1,67 @@
+import {html} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import * as api from '../../../../types/api.js';
+
+import {BaseElement, property} from './base-element.js';
+
+export class NavElementInput extends BaseElement {
+  @property({attribute: true}) title = '';
+  @property({attribute: true}) placeholder = '';
+  @property() endpoint = '';
+  @property() error?: string;
+  @property() private editing = false;
+
+  private startEditing() {
+    this.editing = true;
+  }
+
+  private async onKeyPress(event: KeyboardEvent) {
+    // Check for enter key.
+    if (event.which !== 13) {
+      return;
+    }
+
+    if (!this.endpoint) {
```

Comment:
What is this endpoint and why does it change?

Reply:
Renamed to api endpoint. Going to be re-used for "add person".